### PR TITLE
dyno: resolve more loop index variables

### PR DESCRIPTION
**Pull request by benharsh**
This PR updates dyno's Resolver to resolve index variables for any ``IndexableLoop``, instead of just ``For`` loops.

Testing:
- [x] make test-dyno

### CODE REVIEW

File: compiler/dyno/lib/resolution/Resolver.cpp [hunk 1]
```diff
@@ -2248,7 +2302,7 @@ bool Resolver::enter(const For* loop) {
         loopResults.push_back(std::move(cur.byPostorder));
       }
 
-      auto paramLoop = new ResolvedParamLoop(loop);
+      auto paramLoop = new ResolvedParamLoop(forLoop);
```
Comment (DanilaFe) [inline compiler/dyno/lib/resolution/Resolver.cpp:2305]: Not really part of this PR, but is this ever freed? Should it be?

**Review by DanilaFe — APPROVED** [review]
Looks good to me, but someone else should probably also take a look.

File: compiler/dyno/lib/resolution/Resolver.cpp (hunk 1 above)
Comment (benharsh) [reply compiler/dyno/lib/resolution/Resolver.cpp:2305]: There is definitely a memory-management issue here that warrants further discussion. I think that discussion is a bit complicated and outside of the scope of this PR, so I will merge presently and take the discussion of memory management for ResolvedParamLoop offline.
